apiserver/common/networkingcommon/networkconfigapi_test.go

```diff
@@ -441,6 +438,45 @@ func (s *networkConfigSuite) TestUpdateMachineLinkLayerOpBridgedDeviceMovesAddre
 	c.Assert(err, jc.ErrorIsNil)
 }
 
+func (s *networkConfigSuite) TestUpdateMachineLinkLayerOpReprocessesDevices(c *gc.C) {
+	ctrl := s.setupMocks(c)
+	defer ctrl.Finish()
+
+	hwAddr := "aa:bb:cc:dd:ee:ff"
+
+	s.expectMachine()
+	mExp := s.machine.EXPECT()
+	mExp.AllLinkLayerDevices().Return(nil, nil).Times(2)
+	mExp.AllAddresses().Return(nil, nil).Times(2)
+
+	// Expect the device addition to be attempted twice.
+	mExp.AddLinkLayerDeviceOps(
+		state.LinkLayerDeviceArgs{
+			Name:        "eth0",
+			Type:        "ethernet",
+			MACAddress:  hwAddr,
+			IsAutoStart: true,
+			IsUp:        true,
+		},
+	).Return([]txn.Op{{}, {}}, nil).Times(2)
```

Comment:
I'm trying to make sure I understand this test, but it doesn't make sense to me that it would return 'some Ops to perform' the first time, but 'no Ops' the second time.
If the goal is to tell that we are retrying, shouldn't the Return value be the same both times?

Reply:
The ops returned in this case aren't important - it returns the same slice of zero-value ops both times. The `.Times(2)` is the critical thing here. It fails before this patch, because the device is marked as processed and is no attempt is made to add it when `Build` is reattempted.